### PR TITLE
Feature/download kubectl2

### PR DESCRIPTION
**Pull request by rakelkar**
Draft review to download kubectl instead of getting it from the repo

### CODE REVIEW

File: kubectl.ts [hunk 1]
```diff
@@ -31,26 +31,43 @@ export class KubectlCommand {
...
         this.kubectl.arg(arg);
     }
     exec() {
         this.execCommand();
     }
+    async init() {
+        if (!fs.exists(this.kubectlbinary)) {                
+            tl.debug("downloading kubectl");
+            let curl: ToolRunner = tl.tool("curl");
+            curl.arg("-LO").arg("https://storage.googleapis.com/kubernetes-release/release/v1.5.2/bin/linux/amd64/kubectl");
```
Comment (TsuyoshiUshio) [inline kubectl.ts]: Only the problem is the version number. v1.5.2. This feature has backward compatibility. So currently no problem. However, in the future, we need to add the version column on the Task as optional. And it is not difficult. because we can just insert the version number. https://kubernetes.io/docs/user-guide/prereqs/

File: vss-extension.json~ [hunk 2]
```diff
@@ -10,11 +10,11 @@
       "Public"
   ],
 =======
-  "version": "0.1.25",
-  "name": "Kubernetes extension",
-  "description": "Kubernetes extension with k8s endpoint. Which enable kubectl command",
-  "publisher": "TsuyoshiUshio",
->>>>>>> parent of 00763c9... add support to download kubectl
+  "version": "0.1.45",
+  "name": "Kubernetes Ostrich extension",
```
Comment (dtzar) [inline vss-extension.json~]: Probably want to remove the Ostrich references ;)

Comment (TsuyoshiUshio) [inline vss-extension.json~]: Thanks, I've never noticed. What is 'Ostrich'? Kubernetes release name or something? At least, we need to avoid version problem. I don't want to update everytime. haha.

Comment (rakelkar) [inline vss-extension.json~]: I changed the publish name so that I could publish and iterate :)

Comment (TsuyoshiUshio) [inline vss-extension.json~]: Ah! :)
BTW, according to the answer of the guy, we the download spec can be like this. What do you think?

1. If the KubeBinary is null or empty, the task will download the binary.
2. if the downloadversion is null, it will donwload latest.

```
curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
```

3. if the downloadversion is filled and KubeBinary is null or empty, download the specific binary. it must be for Linux. 
4. the binally is "chmod". However, the vsts-lib.task's chmod doesn't work, so we need to fire chmod by myself.

https://kubernetes.io/docs/user-guide/prereqs/

File: Tests/L0.ts [hunk 3]
```diff
@@ -54,8 +54,7 @@ describe('Kubectl apply Task', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
         tr.run();
         tl.debug(tr.cmdlines);
-        expect(tr.ran("./Tests/kubectl apply -f ./Tests/my-nginx.yml --kubeconfig ./config")).to.be.true;
+        expect(tr.ran("./Tests/kubectl apply -f ./Tests/my-nginx.yml --kubeconfig ./kubeconfig")).to.be.true;
```
Comment (TsuyoshiUshio) [inline Tests/L0.ts]: Why did you change the name? Just curious.

File: kubectl.ts [hunk 4]
```diff
@@ -41,39 +42,42 @@ export class KubectlCommand {
...
-            curl.arg("-LO").arg("https://storage.googleapis.com/kubernetes-release/release/stable.txt");
-            var result = curl.execSync();
-            downloadVersion = result.stdout.toString().trim();
-        }
+        try {                     
+            if (!downloadVersion) {
+                tl.debug("getting kubectl current stable version");
+                let curl: ToolRunner = tl.tool("curl");
+                curl.arg("-L").arg("https://storage.googleapis.com/kubernetes-release/release/stable.txt");
+                var result = curl.execSync();
```
Comment (TsuyoshiUshio) [inline kubectl.ts]: I recommend to use let instead of let. :) var could cause some problem. (Although, a lot of people keep on using var. haha

Comment (TsuyoshiUshio) [inline kubectl.ts]: Also, why did you use execSysn() instead of using await.curl.exec(); Just curious

**Comment by TsuyoshiUshio** [comment]
Overall, great job of @rakelkar ! Thank you for contribution!

**Comment by dtzar** [comment]
Look forward to seeing this update published in the marketplace! :) 🎉

**Comment by TsuyoshiUshio** [comment]
I tried release this. however, it didn't work. I add an issue.  I'll fix and release this. 
https://github.com/TsuyoshiUshio/KubernetesTask/issues/8
